sim/warlock/talents_destruction.go

```diff
@@ -158,20 +166,103 @@ func (warlock *Warlock) registerBackdraft() {
+	if warlock.Talents.BurningEmbers <= 0 {
+		return
+	}
+
+	damageGainPerHit := 0.25 * float64(warlock.Talents.BurningEmbers)
+	spellPowerMultiplier := 0.7 * float64(warlock.Talents.BurningEmbers)
+	baseDamage := warlock.CalcBaseDamage([]float64{0.0, 0.0734, 0.147}[warlock.Talents.BurningEmbers])
+
+	warlock.BurningEmbers = warlock.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 85112},
+		SpellSchool:    core.SpellSchoolFire,
+		ProcMask:       core.ProcMaskSpellDamage,
+		ClassSpellMask: WarlockSpellBurningEmbers,
+
+		ManaCost: core.ManaCostOptions{
+			BaseCost:   0,
+			Multiplier: 1,
 		},
-		OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
-			if spell == warlock.Conflagrate {
-				backdraft.Activate(sim)
-				backdraft.SetStacks(sim, 3)
-			}
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD:      0,
+				CastTime: 0,
+			},
+		},
+
+		DamageMultiplier: 1,
+		CritMultiplier:   warlock.DefaultSpellCritMultiplier(),
+		ThreatMultiplier: 1,
+
+		Dot: core.DotConfig{
+			Aura: core.Aura{
+				Label: "Burning Embers",
+			},
+			NumberOfTicks: 7,
+			TickLength:    time.Second * 1,
+			//?AffectedByCastSpeed: true,
+
+			OnSnapshot: func(sim *core.Simulation, target *core.Unit, dot *core.Dot, isRollover bool) {
+			},
+			OnTick: func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
+				dot.CalcAndDealPeriodicSnapshotDamage(sim, target, dot.OutcomeSnapshotCrit)
+			},
+		},
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			spell.CalcAndDealOutcome(sim, target, spell.OutcomeAlwaysHit)
+			spell.Dot(target).Apply(sim)
 		},
 	})
+
+	core.MakePermanent(
+		warlock.RegisterAura(core.Aura{
+			Label:    "Burning Embers Hidden Aura",
+			ActionID: core.ActionID{SpellID: 85112},
+			OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if !result.Landed() {
+					return
+				}
+				if spell == warlock.SoulFire {
+					warlock.BurningEmbers.Dot(result.Target).Apply(sim)
+				}
+			},
+		}))
+
+	core.MakeProcTriggerAura(&warlock.Unit, core.ProcTrigger{
+		Name:           "Burning Embers",
+		Callback:       core.CallbackOnSpellHitDealt,
+		ClassSpellMask: WarlockSpellSoulFire | WarlockSpellImpFireBolt,
+		Outcome:        core.OutcomeLanded,
+
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			dot := warlock.BurningEmbers.Dot(result.Target)
+			// Max damage is based on the formula which changes per talent point:
+			//1: [(Spell power * 0.7 + 71) / 7]
+			//2: [(Spell power * 1.4 + 141) / 7]
+			maxDamagePerTick := (baseDamage + (dot.Spell.SpellPower() * spellPowerMultiplier)) / 7
+
+			// The damage per tick is then based off the damage of the Imp Firebolt or Soulfire that just hit
+			dot.SnapshotBaseDamage = min(result.Damage*damageGainPerHit, maxDamagePerTick)
+			dot.SnapshotAttackerMultiplier = warlock.BurningEmbers.DamageMultiplier
```

Comment:
`warlock.BurningEmbers.DamageMultiplier * warlock.BurningEmbers.DamageMultiplierAdditive`

Reply:
Not sure what you mean by this?

Reply:
You are transfering multipliers from one spell to another but you're transfering only half of the equation. Snapshot is usually done with both `DamageMultiplier` and `DamageMultiplierAdditive`

Reply:
Ahh I see.